pgtype/timestamp.go

```diff
@@ -104,8 +104,8 @@ func (ts *Timestamp) UnmarshalJSON(b []byte) error {
 	case "-infinity":
 		*ts = Timestamp{Valid: true, InfinityModifier: -Infinity}
 	default:
-		// PostgreSQL uses ISO 8601 for to_json function and casting from a string to timestamptz
-		tim, err := time.Parse(time.RFC3339Nano, *s)
+		// PostgreSQL uses ISO 8601 wihout timezone for to_json function and casting from a string to timestampt
```

Comment:
```suggestion
		// PostgreSQL uses ISO 8601 without timezone for to_json function and casting from a string to timestamptz
```

Reply:
Thanks! I thing I'll extract it as a separate PR - want to keep scope of this one narrow